taildropout.py

```diff
@@ -135,15 +135,16 @@ def forward(self, input: Tensor) -> Tensor:
             device = input.device
 
             linspace = torch.arange(1, n_features + 1, 1, device=device, dtype=type_out)
-            prob_shape = replace_w_ones_except(input.shape, self.dropout_dim) #[1,1,..,n_features]
-            linspace.resize_(prob_shape)
             # self.scale*n_features faster than linspace/n_features
             prob = self.cdf(linspace, self.scale * n_features)
 
+            prob_shape = replace_w_ones_except(input.shape, self.dropout_dim) #[1,1,..,n_features]
+            prob = prob.reshape(prob_shape)
+
             mask_shape = replace_w_ones_except(input.shape, self.batch_dim)
             uniform = torch.rand(mask_shape, device=device, dtype=type_out) # [n_batch,1,1] if input 3d
             mask = prob < uniform         # 43% of cpu cumtime                [n_batch,1,n_features] 
-            mask = mask.type(type_out)    # 30% of cpu cumtime
+            # mask = mask.type(type_out)    # 30% of cpu cumtime
```

Comment:
Keep comment for now as it may have been explicit to get the timing in the first place. Timing probably needs an update too.